http/rest-client-reactive/src/test/java/io/quarkus/ts/http/restclient/reactive/UrlOverrideClientIT.java

```diff
@@ -0,0 +1,127 @@
+
+    /*
+     * Start two application. By default, restClient in app will do HTTP requests against app.
+     * But if we override the URL, it will do requests against app2.
+     * Both apps will be distinguished by value of "ts.quarkus.urlOverride.response" which they return as HTTP response.
+     * The URL override feature overrides base URL, but not HTTP path. So we cannot easily test it using just one app.
+     */
+    @JaegerContainer(expectedLog = "\"Health Check state change\",\"status\":\"ready\"")
+    static final JaegerService jaeger = new JaegerService();
+
+    @QuarkusApplication()
+    static RestService app = new RestService()
+            .withProperties("urlOverride.properties")
+            .withProperty("quarkus.otel.enabled", "true")
+            .withProperty("quarkus.otel.simple", "true")
+            .withProperty("quarkus.application.name", MAIN_SERVICE_NAME)
+            .withProperty("quarkus.otel.exporter.otlp.traces.endpoint", jaeger::getCollectorUrl);
+
+    @QuarkusApplication()
+    static RestService app2 = new RestService()
+            .withProperties("urlOverride.properties")
+            .withProperty("quarkus.http.ssl-port", "8444")
+            .withProperty("ts.quarkus.urlOverride.response", "overridden");
+
+    @BeforeEach
+    public void setTestStartTime() {
+        // store time in same format as jaeger
+        testStartTime = System.currentTimeMillis() * 1000;
+    }
+
+    @Test
+    public void shouldOverrideUrlByString() {
+        testEndpoints("defaultString", "overrideString");
+    }
+
+    @Test
+    public void shouldOverrideUrlByUrl() {
+        testEndpoints("defaultUrl", "overrideUrl");
+    }
+
+    @Test
+    public void shouldOverrideUrlByUri() {
+        testEndpoints("defaultUri", "overrideUri");
+    }
+
+    @Test
+    public void shouldNotOverride() {
+        String response = app.given().get("/testUrlOverride/notOverridden").asString();
+        assertEquals("default", response);
+    }
+
+    @Test
+    public void shouldFail() {
+        // redirect the request to non-existing host which should result in failure to execute the erquest
+        Response response = app.given().get("/testUrlOverride/overrideHost/?host=nonExistingHost");
+        assertEquals(500, response.statusCode(), "Request should fail with code 500");
+        app.logs().assertContains("UnknownHostException: nonExistingHost");
+    }
+
+    private void testEndpoints(String defaultEndpoint, String overrideEndpoint) {
+        int secondAppPort = app2.getURI(Protocol.HTTP).getPort();
+
+        String defaultResponse = app.given().get("/testUrlOverride/" + defaultEndpoint).asString();
+        assertEquals("default", defaultResponse);
+
+        String overriddenResponse = app.given()
+                .get("/testUrlOverride/" + overrideEndpoint + "/?port=" + secondAppPort)
+                .body().asString();
+        assertEquals("overridden", overriddenResponse);
+
+        verifyOtelTraces();
```

Comment:
Just because you set `quarkus.otel.simple` it doesn't mean that sending of traces _must_ be quicker than Quarkus respond arriving here. I think span export is still async. We can keep this and wait if this ever become flaky.